### DISCUSSION TITLE
Speed up pdf files meging

### DISCUSSION THREAD

**Discussion by PeterSlezak** [pikepdf/pikepdf#542] [Q&A] [answered]
Hello,

I just want to check if there is any option/suggestion to speed-up the pdfs merging. I'm using the following code

```
pdf_out_doc = pikepdf.Pdf.open()
for filename in filelist:      
    doc = pikepdf.Pdf.open(file)
    pdf_out_doc.pages.extend(doc.pages)
    doc.close()
pdf_out_doc.save(outpath)
```
The filelist contains file paths to individual pdfs. In my case it's usually about 50-100 files. Individual files contain only text and simple drawings (rectangles, lines) created by reportlab. The size of files together is >100MB (might goes up to 500MB) and the number of pages of all files combined is >100K pages. In found the above code running for ~5 hours to get the concatenated pdf file. Is there any way to speed up the process? 

Thanks in advance for any suggestion

**Comment by jbarlow83** [comment, marked as answer]
It looks like you found a regression that made appending pages take quadratic time:

v8.9.0
10000: 25s
20000: 84s

After fixing this the scaling is linear when adding single pages:
10000: 9.7s
20000: 19.2s
30000: 28.7s

When adding larger sets of pages it's much faster still.

This will be fixed in the next release.

**Comment by jbarlow83** [comment]
v8.10+ should take care of this.

**Reply by PeterSlezak** [reply]
Hi @jbarlow83. I can confirm. My test case now completed in ~4min. 
Thank you for quick update!
